### PR TITLE
Update Getting Started guide with warning about compilation errors

### LINKED ISSUE #6782: psycopg2 can't build
Followup to #3106 (or: "psycopg2: the unfolding").

I was trying to fix a different problem in my developer environment and tried `$ pip install -U -r requirements.txt` and got this error:

~~~
Collecting psycopg2==2.8.3 (from -r requirements/main.txt (line 379))
  Downloading https://files.pythonhosted.org/packages/5c/1c/6997288da181277a0c29bc39a5f9143ff20b8c99f2a7d059cfb55163e165/psycopg2-2.8.3.tar.gz (377kB)
     |████████████████████████████████| 378kB 3.5MB/s 
    ERROR: Command errored out with exit status 1:
     command: /[path redacted]/.virtualenvs/PyPI/bin/python3 -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-0bu02wzi/psycopg2/setup.py'"'"'; __file__='"'"'/tmp/pip-install-0bu02wzi/psycopg2/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base pip-egg-info
         cwd: /tmp/pip-install-0bu02wzi/psycopg2/
    Complete output (23 lines):
    running egg_info
    creating pip-egg-info/psycopg2.egg-info
    writing pip-egg-info/psycopg2.egg-info/PKG-INFO
    writing dependency_links to pip-egg-info/psycopg2.egg-info/dependency_links.txt
    writing top-level names to pip-egg-info/psycopg2.egg-info/top_level.txt
    writing manifest file 'pip-egg-info/psycopg2.egg-info/SOURCES.txt'
    
    Error: pg_config executable not found.
    
    pg_config is required to build psycopg2 from source.  Please add the directory
    containing pg_config to the $PATH or specify the full executable path with the
    option:
    
        python setup.py build_ext --pg-config /path/to/pg_config build ...
    
    or with the pg_config option in 'setup.cfg'.
    
    If you prefer to avoid building psycopg2 from source, please install the PyPI
    'psycopg2-binary' package instead.
    
    For further information please check the 'doc/src/install.rst' file (also at
    <http://initd.org/psycopg/docs/install.html>).
    
    ----------------------------------------
ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
~~~

I don't actually see any instructions to pip install stuff in https://warehouse.readthedocs.io/development/getting-started/  -- we do this all in the Docker container -- so I think this is probably not actually a bug. But in case it is: I see we don't have `pg_config` anywhere listed as a requirement in the Warehouse source; should we?

### PR DESCRIPTION
**Pull request by yeraydiazdiaz**
Fixes #6782 as suggested https://github.com/pypa/warehouse/issues/6782#issuecomment-541328116.

### CODE REVIEW

File: docs/development/getting-started.rst [hunk 1]
```diff
@@ -401,6 +401,23 @@ away your builds + ``docker`` containers and start again:
 ``docker volume rm $(docker volume ls -q --filter dangling=true)``
 
 
+Compilation errors in non-Docker development
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+While Warehouse is designed to be developed in using Docker in certain
```
Comment (di) [inline docs/development/getting-started.rst]:
```suggestion
While Warehouse is designed to be developed in using Docker, in certain
```

File: docs/development/getting-started.rst [hunk 2]
```diff
@@ -401,6 +401,23 @@ away your builds + ``docker`` containers and start again:
...
+situations you may be tempted to install Warehouse's requirements in your
```
Comment (di) [inline docs/development/getting-started.rst]: Tempted?

File: docs/development/getting-started.rst [hunk 3]
```diff
@@ -401,6 +401,23 @@ away your builds + ``docker`` containers and start again:
...
+system of virtual environment. This is discouraged and may result in
```
Comment (di) [inline docs/development/getting-started.rst]:
```suggestion
system or virtual environment. This is discouraged and may result in
```

File: docs/development/getting-started.rst (hunk 2 above)
Comment (yeraydiazdiaz) [reply docs/development/getting-started.rst]: I guess it's too late at this point if they're reading the troubleshooting section 😄 I'll rephrase

**Pull request merged**